app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -147,15 +147,32 @@ def fetch_server_entities
 
       def fetch_availability
         resources_by_metric_id = {}
+        metric_id_by_resource_path = {}
+        @ems.feeds.each do |feed|
+          deployment_status_mt_path = ::Hawkular::Inventory::CanonicalPath.new(
+            :metric_type_id => 'Deployment%20Status~Deployment%20Status', :feed_id => feed
+          )
+          deployment_status_metrics = @ems.inventory_client.list_metrics_for_metric_type(deployment_status_mt_path)
+          deployment_status_metrics.each do |deployment_status_metric|
+            deployment_status_metric_path = ::Hawkular::Inventory::CanonicalPath.parse(deployment_status_metric.path)
+            path = ::Hawkular::Inventory::CanonicalPath.new(:tenant_id    => deployment_status_metric_path.tenant_id,
+                                                            :feed_id      => deployment_status_metric_path.feed_id,
+                                                            :resource_ids => deployment_status_metric_path.resource_ids)
+            metric_id_by_resource_path[path.to_s] = deployment_status_metric.hawkular_metric_id
+          end
+        end
         @data[:middleware_deployments].each do |deployment|
+          # Mark default status for all deployments
+          deployment[:status] = process_availability
           path = ::Hawkular::Inventory::CanonicalPath.parse(deployment[:ems_ref])
           # for subdeployments use it's parent deployment availability.
           path = path.up if path.resource_ids.last.include? CGI.escape('/subdeployment=')
```

Comment:
We should be good here. It's simple 1 lvl tree with the ear in the root and wars or jars in the leaves